### PR TITLE
Create provider with data source

### PR DESCRIPTION
**Pull request by dmattia**
This allows writing terraform like:

```hcl
terraform {
  required_providers {
    pulumi = {
      version = "0.1"
      source  = "hashicorp.com/transcend-io/pulumi"
    }
  }
}

provider "pulumi" {}

data "pulumi_stack_outputs" "stack_outputs" {
  organization = "transcend-io"
  project      = "airgap-telemetry-backend"
  stack        = "dev"
}

output "version" {
  value = data.pulumi_stack_outputs.stack_outputs.version
}

output "stack_outputs" {
  value = data.pulumi_stack_outputs.stack_outputs.stack_outputs
}
```

that will lookup pulumi stack outputs

### CODE REVIEW

**Review by anotherminh — APPROVED** [review]
Seems useful!

**Review by MadDataScience — APPROVED** [review]
I’m pretty rusty at this point, but this all LGTM.

File: pulumi/data_source_stack_outputs.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+func dataSourceStackOutputsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := &http.Client{Timeout: 10 * time.Second}
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	// The ID will be the org, project, and stack names separated with slashes
+	project := d.Get("project").(string)
```
Comment (dipack95) [inline pulumi/data_source_stack_outputs.go:26]: Not a critique, but it's quite funny how Golang does type coercions to me

File: pulumi/provider.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		},
+		ResourcesMap: map[string]*schema.Resource{},
+		DataSourcesMap: map[string]*schema.Resource{
+			"pulumi_stack_outputs": dataSourceStackOutputs(),
+		},
+		ConfigureContextFunc: providerConfigure,
+	}
+}
+
+func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
```
Comment (dipack95) [inline pulumi/provider.go:29]:
```suggestion
func providerConfigure(ctx context.Context, d *schema.ResourceData) (string, diag.Diagnostics) {
```

**Review by dipack95 — APPROVED** [review]
LGTM -- This is pretty neat :)

File: pulumi/provider.go (hunk 2 above)
Comment (dmattia) [reply pulumi/provider.go:29]: I just tried this, but `ConfigureContextFunc` needs this to be an interface I guess

Comment (dipack95) [reply pulumi/provider.go:29]: Odd, considering you're returning a string which should satisfy the interface{} requirements? I guess I'm forgetting how Go types work in these cases

**Pull request merged**